### PR TITLE
ci: release automation

### PR DESCRIPTION
**Pull request by v1v**
### What

Use Buildkite to run the releases interacting with Maven central.

### Why

Jenkins is deprecated hence we use Buildkite for releases, while we use GitHub actions for the CI.

Buildkite will act as a provisioner, pretty much:
- Request a GCP VM
- Access vault
- Run the `release.sh` or `snapshot.sh` scripts
- Notify the result in Slack.

As a consequence, this project will contain all the moving pieces for the release, including the scripts needed for the releases/snapshots.

### Use cases

1) Releases will be triggered manually, using the GitHub actions UI.
2) Snapshots will happen on merge commits automatically and also supported with manually trigger using the GH actions UI

### Tasks

- [x] Create jobs in Buildkite
- [x] Create vault secrets
- [x] Mask passwords/secrets using the Buildkite hooks. 
- [ ] Agree with the @elastic/apm-agent-java team so they are aware of this particular automation.

### CODE REVIEW

File: .buildkite/release.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+agents:
+  provider: "gcp"
```
Comment (reakaleek) [inline .buildkite/release.yml:2]: nit: does this need to run on `gcp`?

from my experience with playing around with buildkite, this provider takes a bit longer to get scheduled than other workers.

**Review by reakaleek — APPROVED** [review]

File: .buildkite/release.yml (hunk 1 above)
Comment (v1v) [reply .buildkite/release.yml:2]: It is needed, since the existing default containers don't have the whole OS setup, such as `gpg`.

Potentially, we could create a container image to support that particular use case, but as it works with `GCP` workers, I'd consider that particular implementation to be done in a follow up. So we can keep migrating other projects.

File: .buildkite/hooks/pre-command [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+export SECRING_FILE=$GNUPGHOME"/secring.gpg"
+gpg --pinentry-mode=loopback --passphrase "$KEYPASS_SECRET" --export-secret-key $KEY_ID_SECRET > "$SECRING_FILE"
+
+echo "--- Configure git context"
+# Configure the committer since the maven release requires to push changes to GitHub
+# This will help with the SLSA requirements.
+git config --global user.email "infra-root+apmmachine@elastic.co"
+git config --global user.name "apmmachine"
+
+echo "--- Install JDK11"
```
Comment (jackshirazi) [inline .buildkite/hooks/pre-command:57]: We should keep this in sync with the agent and use JDK 17, but we can do that in another PR if you want to keep the move to GH and the JDK change separate

File: .buildkite/hooks/pre-command [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+JAVA_URL=https://jvm-catalog.elastic.co/jdk
+JAVA_HOME=$(pwd)/.openjdk11
+JAVA_PKG="$JAVA_URL/latest_openjdk_11_linux.tar.gz"
```
Comment (v1v) [inline .buildkite/hooks/pre-command:60]:
```suggestion
echo "--- Install JDK17"
JAVA_URL=https://jvm-catalog.elastic.co/jdk
JAVA_HOME=$(pwd)/.openjdk17
JAVA_PKG="$JAVA_URL/latest_openjdk_17_linux.tar.gz"
```

File: .buildkite/hooks/pre-command (hunk 2 above)
Comment (v1v) [reply .buildkite/hooks/pre-command]: https://github.com/elastic/ecs-logging-java/pull/197/commits/381c420c9bb1ef5393138ac76fb6aef3803cfc49

**Review by jackshirazi — APPROVED** [review]
